### PR TITLE
Lashup events

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

**Comment by sargun** [comment]
@GoelDeepak

**Comment by GoelDeepak** [comment]
lgtm

**Comment by sargun** [comment]
@aeyakovenko

File: src/lashup_gm_route.erl [hunk 1]
```diff
@@ -258,8 +259,8 @@ start_link() ->
 init([]) ->
   tree_cache = ets:new(tree_cache, [set, named_table, {read_concurrency, true}, {keypos, #tree_cache.root}]),
   vertices = ets:new(vertices, [set, named_table, {keypos, #vertex.node}]),
-  edges = ets:new(edges, [bag, named_table, {keypos, #edge.src}]),
-  {ok, #state{}}.
+  EventsHelper = spawn_link(fun advertise_tree/0),
```
Comment (aeyakovenko) [inline src/lashup_gm_route.erl]: why isn't this a gen server?

Comment (sargun) [inline src/lashup_gm_route.erl]: An entire gen_server seemed a bit heavy. A gen_statem might make more sense.

**Comment by sargun** [comment]
@aeyakovenko @GoelDeepak want to review?

**Comment by GoelDeepak** [comment]
@sargun Could you please comment summarizing this patch?

**Comment by sargun** [comment]
This PR does a few things:

* It changes the way that Lashup BFS is managed, and calculated. If there are too many events that result in regenerating the BFS, it'll throttle the requests
* It uses a gen_event to fan out these events
* It doesn't compress Lashup GM message
* It gets rid of the sliding window metadata in member
* It cleans up the AAE FSMs
* More tests around BFS
* Bump timeouts

**Comment by GoelDeepak** [comment]
lgtm

**Pull request merged**